feg/gateway/services/session_proxy/servicers/session_controller.go

```diff
@@ -81,10 +81,13 @@ func NewCentralSessionController(
 // CreateSession begins a UE session by requesting rules from PCEF
 // and credit from OCS (if RatingGroup is present) and returning them.
 func (srv *CentralSessionController) CreateSession(
-	_ context.Context,
+	ctx context.Context,
 	request *protos.CreateSessionRequest,
 ) (*protos.CreateSessionResponse, error) {
 	glog.V(2).Info("Trying to create session")
+
+	defer orcprotos.ForwardRequestId(ctx)
```

Comment:
just curious, forwardRequestId is sending headers, however here defer is making it send when the function exits, how does this internally work with grpc. Is there any requirement on when headers should be sent or it doesn't really matter.